flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -182,11 +182,17 @@ private static DataType toFlinkPrimitiveType(PrimitiveTypeInfo hiveType) {
 
 		@Override
 		public TypeInfo visit(CharType charType) {
-			if (charType.getLength() > HiveChar.MAX_CHAR_LENGTH) {
-				throw new CatalogException(
-						String.format("HiveCatalog doesn't support char type with length of '%d'. " +
-									"The maximum length is %d",
+			// Flink treats string literal UDF parameters as CHAR. Such types may have precisions not supported by
+			// Hive, e.g. CHAR(0). Promote it to STRING in such case if we're told not to check precision.
+			if (charType.getLength() > HiveChar.MAX_CHAR_LENGTH || charType.getLength() < 1) {
```

Comment:
Also modify varchar?

Reply:
I'm not sure whether that makes sense because currently we don't have a use case to involve a `VARCHAR(0)`.

Even for CHAR, Flink also has some logic to [verify the length](https://github.com/apache/flink/blob/master/flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/CharType.java#L67) is > 0. So having this CHAR(0) as UDF parameter type seems a little inconsistent in the first place.

Reply:
No, you can not have this assumption.
Maybe there is a VarChar with zero precision in somewhere.